src/Offer/Offer.php

```diff
@@ -637,9 +640,13 @@ private function updateImageAllowed(AbstractUpdateImage $updateImageCommand)
      */
     public function removeImage(Image $image)
     {
-        if ($this->images->contains($image)) {
+        // Find the image based on UUID inside the internal state.
+        // Use the image from the internal state.
+        $existingImage = $this->images->findImageByUUID($image->getMediaObjectId());
+
+        if ($existingImage && $this->images->contains($existingImage)) {
```

Comment:
Ik vraag me af of de `$this->images->contains()` hier nog wel nodig is @LucWollants ? Je haalt met `findImageByUUID()` de afbeelding uit de collectie (als er een met dat uuid bestaat), dus dan zou de `contains()` altijd `true` moeten zijn tenzij `$existingImage` gelijk aan `null` is, maar dan geraak je sowieso niet verder dan `if ($existingImage ...`.

Reply:
Inderdaad, die test kan simpler. Goed gezien!